ngraph/core/include/ngraph/validation_util.hpp

```diff
@@ -230,10 +230,33 @@ namespace ngraph
                            std::uint64_t tensor_rank,
                            std::int64_t axis_range_min,
                            std::int64_t axis_range_max);
+    /// \brief Checks if shape_1[ind_1] and shape_2[ind_2] could be extracted (rank may be dynamic
+    /// or less than indices) then checks if extracted values are equal (intersection is not empty
+    /// for dynamic dimensions)
+    /// \return true if both checks pass otherwise false
+    NGRAPH_API
+    bool dims_are_equal(const PartialShape& shape_1,
+                        const Dimension::value_type& ind_1,
+                        const PartialShape& shape_2,
+                        const Dimension::value_type& ind_2);
+
+    /// \brief Checks if shape[ind] could be extracted (rank may be dynamic or less than ind_1)
+    /// then checks if extracted value is equal to provided dimension (intersection is not empty for
+    /// dynamic dimensions)
+    /// \return true if both checks pass otherwise false
+    NGRAPH_API
+    bool dims_are_equal(const PartialShape& shape,
+                        const Dimension::value_type& ind,
+                        const Dimension& value);
```

Comment:
we exporting a lot of stuff in `validation_utils.hpp` which actually used internally. Maybe such methods can be put to `src` folder? Or it's planned to be used in transformations / plugins for their internal ops supporting dynamism?